python/templates/macros/declarations.jinja2

```diff
@@ -97,7 +97,7 @@
   bool operator!=(const {{ inverse_type }}& other) const { return !(*this == other); }
 
   // less comparison operator, so that objects can be e.g. stored in sets.
-  bool operator<(const {{ full_type }}& other) const { return m_obj < other.m_obj; }
+  bool operator<(const {{ full_type }}& other) const { return podio::detail::getOrderKey(*this) < podio::detail::getOrderKey(other); }
```

Comment:
Is this strictly necessary? Wondering whether the compiler sees through all of this wrt optimizations.

Reply:
For datatypes this is not necessary, I just thought it could be easier to maintain if the same mechanism was used for both datatypes and interfaces. I'll have to take a look at the cost